src/sentry/api/endpoints/organization_member_details.py

```diff
@@ -56,7 +58,11 @@
 )
 
 
-def get_allowed_roles(request, organization, member=None):
+def get_allowed_roles(
+    request: Request,
+    organization: Organization,
+    member: OrganizationMember | None = None,
```

Comment:
```suggestion
    member: Optional[OrganizationMember] = None,
```

Reply:
We use `| None` a fair bit around the codebase

Reply:
Not sure if there's rule for this. I prefer `| None` because it's fewer characters

Reply:
So for me `X | None` is pretty clear but it breaks down when you do:
- `Any | None` instead of `Optional[Any]`
- `X | Y | None` instead of `Optional[X|Y]`
- use `TypedDict`:
```
class X(TypedDict):
    y: str | None  # Value must be a type
```
Once we decide it should be easy to enforce with linting rules.